src/main/java/com/commercetools/sync/inventories/SkuKeyTuple.java

```diff
@@ -0,0 +1,117 @@
+package com.commercetools.sync.inventories;
+
+import io.sphere.sdk.inventory.InventoryEntry;
+import io.sphere.sdk.inventory.InventoryEntryDraft;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+/**
+ * Helper class that holds {@code sku} of {@link io.sphere.sdk.inventory.InventoryEntry} and the {@code key} of the
+ * {@link io.sphere.sdk.channels.Channel} it belongs to. Presence of sku and key in created instance is similar as for
+ * inventory entry resource: supply channel key can be {@code null} value since it is optional in inventory entry,
+ * but sku has to be not null neither not empty.
+ * Can be used as key in Maps.
+ */
+final class SkuKeyTuple {
+
+    static final String SKU_NOT_SET_MESSAGE = "Can't create SkuKeyTuple instance of inventory entry with no SKU";
+
+    private final String sku;
+
+    @Nullable
+    private final String key;
+
+    private SkuKeyTuple(final String sku, final String key) {
+        this.sku = sku;
+        this.key = key;
+    }
+
+    /**
+     * Returns new {@link SkuKeyTuple} of given {@link InventoryEntry}. If contains
+     * {@link io.sphere.sdk.models.Reference} to supply channel, then reference should be expanded, so that key is
+     * taken from it. If reference is null or is not expanded it's result in {@code null} key value.
+     * @param existingEntry must have sku set and not empty
+     * @return new instance of {@link SkuKeyTuple}
+     * @throws IllegalArgumentException when sku of {@code existingEntry} is null or empty string
+     */
+    static SkuKeyTuple of(@Nonnull final InventoryEntry existingEntry) {
```

Comment:
please as a convention let's always use `Old..` instead of `existing..` everywhere in the sync library. so avoid words like existing, legacy..